mlflow/store/db_artifact_repo.py

```diff
@@ -0,0 +1,304 @@
+                )
+            session.add(artifact)
+            session.flush()
+            return str(artifact.artifact_id)
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        """
+        Log the files in the specified local directory as artifacts, optionally taking
+        an ``artifact_path`` to place them in within the run's artifacts.
+
+        :param local_dir: Directory of local artifacts to log
+        :param artifact_path: Directory within the run's artifact directory in which to log the
+                              artifacts
+        """
+        if artifact_path:
+            artifact_path = os.path.normpath(artifact_path)
+
+        with self.ManagedSessionMaker() as session:
+            for subdir_path, _, files in os.walk(local_dir):
+                relative_path = _relative_path_local(local_dir, subdir_path)
+
+                if artifact_path is None:
+                    db_subdir_path = relative_path if relative_path else ""
+                else:
+                    db_subdir_path = os.path.join(artifact_path, relative_path) \
+                        if relative_path else artifact_path
+
+                for each_file in files:
+                    source = os.path.join(subdir_path, each_file)
+                    if db_subdir_path == "":
+                        artifact = SqlArtifact(
+                            artifact_name=each_file, group_path=self.root,
+                            artifact_content=open(source, "rb").read(),
+                            artifact_initial_size=os.path.getsize(source)
+                        )
+                    else:
+
+                        artifact = SqlArtifact(
+                            artifact_name=each_file,
+                            group_path=os.path.join(self.root, db_subdir_path),
+                            artifact_content=open(source, "rb").read(),
+                            artifact_initial_size=os.path.getsize(source)
+                        )
+                    session.add(artifact)
+        session.flush()
+
+    def list_artifacts(self, path):
+        """
+        Return all the artifacts for this run_id directly under path. If path is a file, returns
+        an empty list. Will error if path is neither a file nor directory.
+
+        :param path: Relative source path that contains desired artifacts
+
+        :return: List of artifacts as FileInfo listed directly under path.
+        """
+        path = os.path.normpath(path)
+        with self.ManagedSessionMaker() as session:
+            regex = os.path.join(path, '%')
+            return [artifact.to_file_info()
+                    for artifact in
+                    session.query(SqlArtifact).filter(
+                        or_(SqlArtifact.group_path.like(
+                            os.path.normpath(os.path.join(self.root, regex))),
+                            SqlArtifact.group_path == os.path.normpath(
```

Comment:
Is it possible to refactor this so that only the `like` clause is needed? Combined with the change suggested above, this might look something like:

```
        path = os.path.join(self.root, path) if path is not None else self.root
        with self.ManagedSessionMaker() as session:
            return [artifact.to_file_info()
                    for artifact in
                    session.query(SqlArtifact).filter(
                        SqlArtifact.group_path.like(
                            path + "%"))]
```

Reply:
This does not work in various edge cases. For example, one might ask for artifacts under /artifacts/path and get also artifact under /artifacts/path2